modules/sor/sorV2/lib/utils/helpers.ts

```diff
@@ -77,41 +79,53 @@ export function getOutputAmount(paths: PathWithAmount[]): TokenAmount {
     return amounts.reduce((a, b) => a.add(b));
 }
 
-export function getHookState(pool: any): HookState | undefined {
-    if (pool.hook === undefined || pool.hook === null) {
+export function getHookState(pool: PrismaPoolAndHookWithDynamic): HookState | undefined {
```

Comment:
This is a great change. I recall running into build issues when trying it. But it seems like back then I was not aware of the existing HookData type, which would have allowed me to assign the pool the PrismaPoolAndHookWithDynamic type (but still stick to using the name). The recently added `GqlHookType` by franz is a great addition.

Reply:
Just FYI, hook.name is being deprecated in the favor of hookType